src/sentry/integrations/slack/message_builder/issues.py

```diff
@@ -445,6 +447,31 @@ def __init__(
         self.is_unfurl = is_unfurl
         self.skip_fallback = skip_fallback
         self.notes = notes
+        self.issue_summary: dict[str, Any] | None = None
+
+    def fetch_issue_summary(self) -> dict[str, Any] | None:
+        """
+        Try to fetch an issue summary with a timeout of 5 seconds.
+        Returns the summary data if successful, None otherwise.
+        """
+        if self.group.issue_category != GroupCategory.ERROR:
+            return None
+        if not features.has("organizations:gen-ai-features", self.group.organization):
+            return None
+        if not features.has("projects:trigger-issue-summary-on-alerts", self.group.project):
+            return None
+
+        try:
+            with concurrent.futures.ThreadPoolExecutor() as executor:
+                future = executor.submit(get_issue_summary, self.group)
+                summary_result, status_code = future.result(timeout=5)
+
+                if status_code == 200:
+                    return summary_result
+                return None
+        except (concurrent.futures.TimeoutError, Exception) as e:
+            logger.exception("Error generating issue summary: %s", e)
```

Comment:
Do we want metrics to track these failures?

Reply:
Is a span and the Sentry issue generated from this sufficient? I'm not too knowledgeable about other ways to track metrics here

Reply:
That should be fine for now. If we want more granular tracking of things like duration over time, success/failure count, we tend to wrap calls like these in metrics decorators like this: https://github.com/getsentry/sentry/blob/c284a68ce24b67e54595ec9bfeb3833ec426a41a/src/sentry/tasks/groupowner.py#L57

This is totally optional though.